File: Examples/basic-sample/basic-sample/Images.xcassets/ProfileViewController.m
//
//  ProfileViewController.m
//  basic-sample
//
//  Created by Martin Gontovnikas on 30/09/14.
//  Copyright (c) 2014 Auth0. All rights reserved.
//

#import "ProfileViewController.h"
#import "Application.h"
#import <SDWebImage/UIImageView+WebCache.h>
#import <Lock/Lock.h>
#import <SimpleKeychain/A0SimpleKeychain.h>
#import <AFNetworking/AFHTTPRequestOperation.h>
#import <libextobjc/EXTScope.h>

@interface ProfileViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *profileImage;
@property (weak, nonatomic) IBOutlet UILabel *welcomeLabel;

- (IBAction)callAPI:(id)sender;

@end

@implementation ProfileViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    A0SimpleKeychain *keychain = [[Application sharedInstance] store];
    A0UserProfile *profile = [NSKeyedUnarchiver unarchiveObjectWithData:[keychain dataForKey:@"profile"]];
    [self.profileImage sd_setImageWithURL:profile.picture];
    self.welcomeLabel.text = [NSString stringWithFormat:NSLocalizedString(@"Welcome %@!", nil), profile.name];
}

- (void)callAPI:(id)sender {
    NSString *baseURLString = [[NSBundle mainBundle] infoDictionary][@"SampleAPIBaseURL"];
    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:baseURLString]];
    AFHTTPRequestOperation *operation = [[AFHTTPRequestOperation alloc] initWithRequest:request];
    @weakify(self);
    [operation setCompletionBlockWithSuccess:^(AFHTTPRequestOperation *operation, id responseObject) {
        @strongify(self);
        [self showMessage:@"We got the secured data successfully"];
    } failure:^(AFHTTPRequestOperation *operation, NSError *error) {
        @strongify(self);
        [self showMessage:@"Please download the API seed so that you can call it."];
    }];
    [operation start];
}

- (void)showMessage:(NSString *)message {
    UIAlertView *alert = [[UIAlertView alloc] initWithTitle:message message:nil delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil];
    [alert show];
}

@end

File: Examples/cognito-sample-swift/SwiftSample/ProfileViewController.swift
//
//  ProfileViewController.swift
//  SwiftSample
//
//  Created by Hernan Zalazar on 10/2/14.
//  Copyright (c) 2014 Auth0. All rights reserved.
//

import UIKit

class ProfileViewController: UIViewController {


    @IBOutlet var profileImage: UIImageView?
    @IBOutlet var welcomeLabel: UILabel?
    @IBOutlet weak var textValue: UITextField!
    
    
    var dataset : AWSCognitoDataset?

    override func viewDidLoad() {
        super.viewDidLoad()
        let keychain = MyApplication.sharedInstance.keychain
        let profileData:NSData! = keychain.dataForKey("profile")
        let profile:A0UserProfile = NSKeyedUnarchiver.unarchiveObjectWithData(profileData) as A0UserProfile
        self.profileImage?.sd_setImageWithURL(profile.picture)
        self.welcomeLabel?.text = "Welcome \(profile.name)!"
        
        let cognitoSync = AWSCognito.defaultCognito()
        dataset = cognitoSync.openOrCreateDataset("MainDataset")
        
        getAndShowData()
    }
    
    func getAndShowData() {
        MBProgressHUD.showHUDAddedTo(self.view, animated: true)
        dataset!.synchronize().continueWithBlock { (task) -> AnyObject! in
            dispatch_async(dispatch_get_main_queue(), { () -> Void in
                self.textValue.text = self.dataset!.stringForKey("value")
                MBProgressHUD.hideHUDForView(self.view, animated: true)
            })
            return nil
        }
    }
    
    
    @IBAction func setStoredValue(sender: AnyObject) {
        dataset!.setString(self.textValue.text, forKey: "value")
        dataset!.synchronize().continueWithBlock { (task) -> AnyObject! in
            return nil
        }
    }
    
    
    @IBAction func getStoredValue(sender: AnyObject) {
        getAndShowData()
    }

}
